File: benchmark/string_equals_2.php
<?php
$start = microtime(true);

$abc1 = "abc";
$abc2 = "abc";
$l1 = "a slightly longer string";
$l2 = "a slightly longer string";
$ll1 = "a significantly longer string but still not overwhelmingly long string";
$ll2 = "a significantly longer string but still not overwhelmingly long string";
$c1 = "changed one character";
$c2 = "changed !ne character";
$numstr = "123";
$al1 = "a slightly longer string";
$al2 = "a slightly longer string!";


$count = 0;
for ($i=0; $i<1000000; $i++) {
  if ($abc1 == $abc2) $count = $count + 1;
  if ($l1 ==      $l2) $count = $count + 1;
  if ($ll1 == $ll2) $count = $count + 1;

  if ("" == $abc1) $count = $count + 1;
  if ($abc1 == "abcd") $count = $count + 1;
  if ($c1 == $c2) $count = $count + 1;
  if ($numstr == 123) $count = $count + 1;
  if ($al1 ==      $al2) $count = $count + 1;
  if ($al1 ==      "a slightly longer strinh") $count = $count + 1;
  if ($ll1 ==       "another") $count = $count + 1;
}

$start = microtime(true) -$start;
echo "$count\r\n";
echo "Elapsed: $start";
?>

File: benchmark/string_equals_2.lua
local start = os.clock()

local abc1 = "abc";
local  abc2 = "abc";
local  l1 = "a slightly longer string";
local  l2 = "a slightly longer string";
local  ll1 = "a significantly longer string but still not overwhelmingly long string";
local  ll2 = "a significantly longer string but still not overwhelmingly long string";
local  c1 = "changed one character";
local  c2 = "changed !ne character";
local  numstr = "123";
local  al1 = "a slightly longer string";
local  al2 = "a slightly longer string!";


local  count = 0
for i = 1, 1000000 do
  if (abc1 == abc2) then count = count + 1 end
  if (l1 ==      l2) then count = count + 1 end
  if (ll1 == ll2) then count = count + 1 end

  if ("" == abc1) then count = count + 1 end
  if (abc1 == "abcd") then count = count + 1 end
  if (c1 == c2) then count = count + 1 end
  if (numstr == 123) then count = count + 1 end
  if (al1 ==      al2) then count = count + 1 end
  if (al1 ==      "a slightly longer strinh") then count = count + 1 end
  if (ll1 ==       "another") then count = count + 1 end
end

start = os.clock()-start
print(count)
print('Elapsed: ' .. start)
